File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKIX/AttributeGeneralNames.cs
using System;
using System.Runtime.Serialization;

//	AttrributeGeneralNames ::= SEQUENCE {
//		issuerName						GeneralNames		OPTIONAL,
//		baseCertificateID [0] IMPLICIT	IssuerSerial		OPTIONAL,
//		objectDigestInfo  [1] IMPLICIT	ObjectDigestInfo	OPTIONAL
//}

namespace Aladdin.ASN1.ISO.PKIX
{
	[Serializable]
	public class AttrributeGeneralNames : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<GeneralNames		>().Factory(), Cast.O,	Tag.Any			), 
			new ObjectInfo(new ObjectCreator<IssuerSerial		>().Factory(), Cast.O,	Tag.Context(0)	), 
			new ObjectInfo(new ObjectCreator<ObjectDigestInfo	>().Factory(), Cast.O,	Tag.Context(1)	), 
		}; 
		// конструктор при сериализации
        protected AttrributeGeneralNames(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public AttrributeGeneralNames(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public AttrributeGeneralNames(GeneralNames issuerName, IssuerSerial baseCertificateID, 
			ObjectDigestInfo objectDigestInfo) : 
			base(info, issuerName, baseCertificateID, objectDigestInfo) {}

		public GeneralNames		IssuerName			{ get { return (GeneralNames	)this[0]; } }
		public IssuerSerial		BaseCertificateID	{ get { return (IssuerSerial	)this[1]; } } 
		public ObjectDigestInfo	ObjectDigestInfo	{ get { return (ObjectDigestInfo)this[2]; } }
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkix/AttributeGeneralNames.java
package aladdin.asn1.iso.pkix;

import aladdin.asn1.*; 
import java.io.*;

//	AttrributeGeneralNames ::= SEQUENCE {
//		issuerName						GeneralNames		OPTIONAL,
//		baseCertificateID [0] IMPLICIT	IssuerSerial		OPTIONAL,
//		objectDigestInfo  [1] IMPLICIT	ObjectDigestInfo	OPTIONAL
//}

public final class AttributeGeneralNames extends Sequence<IEncodable>
{
    private static final long serialVersionUID = -8675581610622444090L;
    
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(GeneralNames    .class).factory(), Cast.O,	Tag.ANY			), 
		new ObjectInfo(new ObjectCreator(IssuerSerial    .class).factory(), Cast.O,	Tag.context(0)	), 
		new ObjectInfo(new ObjectCreator(ObjectDigestInfo.class).factory(), Cast.O,	Tag.context(1)	), 
	}; 
	// конструктор при раскодировании
	public AttributeGeneralNames(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public AttributeGeneralNames(GeneralNames issuerName, 
		IssuerSerial baseCertificateID, ObjectDigestInfo objectDigestInfo) 
	{
		super(info, issuerName, baseCertificateID, objectDigestInfo); 
	}
	public final GeneralNames		issuerName		 () { return (GeneralNames	  )get(0); }
	public final IssuerSerial		baseCertificateID() { return (IssuerSerial	  )get(1); } 
	public final ObjectDigestInfo	objectDigestInfo () { return (ObjectDigestInfo)get(2); }
}
